qcodes/instrument/parameter.py

```diff
@@ -650,7 +600,7 @@ def set_wrapper(value: ParamDataType, **kwargs: Any) -> None:
 
                     set_function(raw_val_step, **kwargs)
 
-                    self._update_cache_with(value=val_step,
+                    self.cache._update_with(value=val_step,
```

Comment:
Maybe we could also include this into the `cache.set` and invoke `set_function` with `cache.raw_value`.

Reply:
similar to the `validate` kwarg suggestion, as agreed offline, let's not do this such that the `set_wrapper` method is kept clean/better-structured/more-explicit.